sklearn/neighbors/_binary_tree.pxi

```diff
@@ -643,30 +639,29 @@ cdef class NodeHeap:
 
         heap[i].val < min(heap[2 * i + 1].val, heap[2 * i + 2].val)
     """
-    cdef cnp.ndarray data_arr
-    cdef NodeHeapData_t[::1] data
+    cdef NodeHeapData_t[:] data
     cdef ITYPE_t n
 
     def __cinit__(self):
```

Comment:
```suggestion
    def __cinit__(self):
        # A one-elements array is used as a placeholder to prevent
        # any problem due to potential access to this attribute
        # (e.g. assigning to NULL or a to value in another segment).
```